src/main/java/dk/dbc/connector/culr/CulrConnector.java

```diff
@@ -113,4 +120,46 @@ public CulrResponse createAccount(String agencyId, UserIdValueAndType userCreden
     public String getEndpoint() {
         return endpoint;
     }
+
+    public int getCacheSize() {
+        return cache.size();
+    }
+
+    private GetAccountsByAccountIdResponse getAccountFromProviderNoCache(String agencyId, UserIdValueAndType userCredentials,
+                                                                         AuthCredentials authCredentials) throws CulrConnectorException {
+        try {
+            return Failsafe.with(retryPolicy).get(() -> proxy.getAccountFromProvider(agencyId, userCredentials, authCredentials));
+        } catch (RuntimeException e) {
+            throw new CulrConnectorException(e.getMessage(), e);
+        }
+    }
+
+    private static class CacheKey {
+        private final String agencyId;
+        private final UserIdTypes userIdType;
+        private final String userIdValue;
+        private final String userIdAut;
+        private final String groupIdAut;
+
+        public CacheKey(String agencyId, UserIdValueAndType userCredentials, AuthCredentials authCredentials) {
+            this.agencyId = agencyId;
+            userIdType = userCredentials.getUserIdType();
+            userIdValue = userCredentials.getUserIdValue();
+            userIdAut = authCredentials.getUserIdAut();
+            groupIdAut = authCredentials.getGroupIdAut();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            CacheKey cacheKey = (CacheKey) o;
+            return userIdValue.equals(cacheKey.userIdValue) && agencyId.equals(cacheKey.agencyId) && userIdType == cacheKey.userIdType && userIdAut.equals(cacheKey.userIdAut) && groupIdAut.equals(cacheKey.groupIdAut);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(agencyId, userIdType, userIdValue, userIdAut, groupIdAut);
+        }
+    }
```

Comment:
Det er egentlig ret smart. Ofte sammensætter vi en String som nøgle, men det er da meget mere elegant med et objekt :)

Reply:
Ude på det store net, kalder man sammensatte strings som map keys for antipatterns, men jo nogle gange er det forbandet bekvemt.